### DISCUSSION TITLE
Error in run workflow

### DISCUSSION THREAD

**Discussion by quiple** [stefanzweifel/git-auto-commit-action#220] [Help]
I used this workflow:

```yaml
name: compress

on:
  push:
    paths:
      - '**.ttf'
  workflow_dispatch:

jobs:
  compress:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-python@v3
        with:
          python-version: '3.x'
      - run: |
          pip install fonttools[woff]
          fonttools ttLib.woff2 compress DosKor.ttf
          fonttools ttLib.woff2 compress WinKor.ttf
      - uses: stefanzweifel/git-auto-commit-action@v4
```

and It made this error:

```bash
Started: bash /home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/entrypoint.sh
INPUT_REPOSITORY value: .
INPUT_STATUS_OPTIONS: 
INPUT_BRANCH value: 
Your branch is up to date with 'origin/main'.
INPUT_ADD_OPTIONS: 
INPUT_FILE_PATTERN: .
INPUT_COMMIT_OPTIONS: 
INPUT_COMMIT_USER_NAME: github-actions[bot]
INPUT_COMMIT_USER_EMAIL: github-actions[bot]@users.noreply.github.com
INPUT_COMMIT_MESSAGE: Apply automatic changes
INPUT_COMMIT_AUTHOR: quiple <quiple@users.noreply.github.com>
[main 7f6fe82] Apply automatic changes
 Author: quiple <quiple@users.noreply.github.com>
 2 files changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 DosKor.woff2
 create mode 100644 WinKor.woff2
INPUT_TAGGING_MESSAGE: 
No tagging message supplied. No tag will be added.
INPUT_PUSH_OPTIONS: 
remote: Permission to quiple/windoskor.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/quiple/windoskor/': The requested URL returned error: 403
Error: Invalid status code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1022:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5) {
  code: 128
}
Error: Invalid status code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1022:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
```

Did I do something wrong? What causes can you expect?

**Comment by stefanzweifel** [comment]
Looks like a permission error.

> remote: Permission to quiple/windoskor.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/quiple/windoskor/': The requested URL returned error: 403

I see that you've triggered the workflow by using `workflow_dispatch`: https://github.com/quiple/windoskor/runs/6464609466?check_suite_focus=true
Maybe the error is related to `actions/checkout@v3`? Could you maybe downgrade to `actions/checkout@v2` and give it another go?

The problem seems to be that the git authentication setup by `actions/checkout` is somehow broken once `git-auto-commit-action` is executed.

**Reply by quiple** [reply]
<https://github.com/quiple/windoskor/runs/6465313323?check_suite_focus=true>

I downgrade to `actions/checkout@v2` and gave it another go, but it seems to be the same result.

**Reply by ZeroRin** [reply]
Maybe because you did not give write permission to github action? It defaults to read only in the early days iirc.
Configure that in the repository's settings-actions-general

**Reply by lorensr** [reply]
I have this issue as well, both on checkout v2 and v3. Haven't changed default settings, which includes write:

<img width="1016" alt="image" src="https://user-images.githubusercontent.com/251288/194982049-cfaa1990-4dcc-4f40-9c4c-ccb233745689.png">

Made a simple repro. 

yml file:

https://github.com/lorensr/test-gha/blob/main/.github/workflows/ci.yml

v2 failure:

https://github.com/lorensr/test-gha/actions/runs/3223648226/jobs/5273946735

v3 failure:

https://github.com/lorensr/test-gha/actions/runs/3223617767/jobs/5273886197

**Reply by stefanzweifel** [reply]
@lorensr In your particular case you're using the action in forks. This is quite the important detail, as pushing to forks is quite complicated with Actions..

I wanted to write, that you need to use the `pull_request_target`, but you're already doing this. 🤔 
You probably already saw the "Use in forks from public repositories" section under the ["Advanced Uses"](https://github.com/stefanzweifel/git-auto-commit-action#advanced-uses)?

Can't give an answer on how to solve this issue, other than mentioning, if you can maybe run your Workflow on your default branch?

> If you use this Action in combination with a linter/fixer, it's easier if you run the Action on push on your main-branch.

**Reply by lorensr** [reply]
> You probably already saw the "Use in forks from public repositories" section under the ["Advanced Uses"](https://github.com/stefanzweifel/git-auto-commit-action#advanced-uses)?

Yes, I've been trying to get the Action to run in the base repo, not the fork.

>  if you can maybe run your Workflow on your default branch?

Yeah, I haven't tried that—trying to avoid extra commits on `main`

Thanks for the response! I think for now I'll disable the commit step on forks. (PR opener will have to manually run prettier and commit)

**Reply by perspilling** [reply]
I have the same problem, and I'm running it on the `main` branch. I'm triggering it manually.
```
      - name: Update CHANGELOG
        uses: stefanzweifel/changelog-updater-action@v1.6.0
        with:
          latest-version: ${{ steps.build_changelog.outputs.toTag }}
          release-notes: ${{ steps.build_changelog.outputs.changelog }}

      - name: Commit updated CHANGELOG
        uses: stefanzweifel/git-auto-commit-action@v4.15.1
        with:
          branch: main
          commit_message: Update CHANGELOG
          file_pattern: CHANGELOG.md
```

**Reply by perspilling** [reply]
Ah, my bad! I had specified the wrong branch name. My main branch is called `master` 🤦‍♂️
